userspace/libscap/scap_procs.c

```diff
@@ -258,21 +258,23 @@ int32_t scap_proc_fill_info_from_stats(scap_t *handle, char* procdirname, struct
 		return SCAP_FAILURE;
 	}
 
-	if(fgets(line, sizeof(line), f) == NULL)
+	size_t ssres = fread(line, 1, sizeof(line) - 1, f);
+	if(ssres < 0)
```

Comment:
Unless I'm mistaken, I don't think `fread` can return a negative value (from the man page: `If an error occurs, or the end of the file is reached, the return value is a short item count (or zero).`). Since you're reading a virtual file from `/proc` that you have already opened, I don't consider this weak error checking a major problem.